### PR TITLE
feat: add obs frame

### PR DESCRIPTION
**Pull request by deer-hang**

### CODE REVIEW

File: cmd/obs-csi-plugin/main.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (chengxiangdong) [inline cmd/obs-csi-plugin/main.go:2]: s/2020/2022

File: pkg/obs/controllerserver.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (chengxiangdong) [inline pkg/obs/controllerserver.go:2]: ditto

File: pkg/obs/driver.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (chengxiangdong) [inline pkg/obs/driver.go:2]: ditto

File: pkg/obs/identityserver.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (chengxiangdong) [inline pkg/obs/identityserver.go:2]: ditto

File: pkg/obs/nodeserver.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (chengxiangdong) [inline pkg/obs/nodeserver.go:2]: ditto

File: pkg/obs/server.go [hunk 6]
```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (chengxiangdong) [inline pkg/obs/server.go:2]: s/2017/2022

File: cmd/obs-csi-plugin/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,7 @@
+FROM centos:7.6.1810
```
Comment (chengxiangdong) [inline cmd/obs-csi-plugin/Dockerfile:1]: may be you can add this file later

File: cmd/obs-csi-plugin/build-obsfs.sh [hunk 8]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
```
Comment (chengxiangdong) [inline cmd/obs-csi-plugin/build-obsfs.sh:1]: may be you can add this file later

File: cmd/obs-csi-plugin/main.go [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/utils/metadatas"
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/utils/mounts"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/klog/v2"
+)
+
+var (
+	endpoint    string
+	nodeID      string
```
Comment (chengxiangdong) [inline cmd/obs-csi-plugin/main.go:36]: pls delete `nodeID`, we don't use it anymore

File: cmd/obs-csi-plugin/main.go [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+			d.SetupDriver(mount, metadata)
+			d.Run()
+		},
+	}
+
+	cmd.Flags().AddGoFlagSet(flag.CommandLine)
+
+	cmd.PersistentFlags().StringVar(&endpoint, "endpoint", "unix://tmp/csi.sock", "CSI endpoint")
+
+	cmd.PersistentFlags().StringVar(&nodeID, "nodeid", "", "node id")
```
Comment (chengxiangdong) [inline cmd/obs-csi-plugin/main.go:83]: ditto

File: cmd/obs-csi-plugin/main.go [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/utils/mounts"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/klog/v2"
+)
+
+var (
+	endpoint    string
+	nodeID      string
+	cloudconfig string
```
Comment (chengxiangdong) [inline cmd/obs-csi-plugin/main.go:37]: s/cloudconfig/cloudConfig

File: pkg/obs/driver.go [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/utils/mounts"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog/v2"
+)
+
+const (
+	driverName  = "obs.csi.huaweicloud.com"
+	topologyKey = "topology." + driverName + "/zone"
+	version     = "1.0.0"
```
Comment (chengxiangdong) [inline pkg/obs/driver.go:33]: the version of the plugin should be set at build time, please refer to
https://github.com/huaweicloud/huaweicloud-csi-driver/blob/master/pkg/evs/driver.go#L48

File: pkg/obs/driver.go [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+	cscap []*csi.ControllerServiceCapability
+	nscap []*csi.NodeServiceCapability
+}
+
+func NewDriver(nodeID, endpoint string, cloud *config.CloudCredentials) *Driver {
+	klog.Infof("Driver: %v version: %v", driverName, version)
+
+	d := &Driver{}
+	d.name = driverName
+	d.nodeID = nodeID
```
Comment (chengxiangdong) [inline pkg/obs/driver.go:57]: pls remove it, we do not use it anymore

File: pkg/obs/driver.go [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+	d.endpoint = endpoint
+	d.cloud = cloud
+
+	d.AddControllerServiceCapabilities(
+		[]csi.ControllerServiceCapability_RPC_Type{
+			csi.ControllerServiceCapability_RPC_CREATE_DELETE_VOLUME,
+			csi.ControllerServiceCapability_RPC_LIST_VOLUMES,
+			csi.ControllerServiceCapability_RPC_GET_VOLUME,
+		})
+	d.AddVolumeCapabilityAccessModes([]csi.VolumeCapability_AccessMode_Mode{
```
Comment (chengxiangdong) [inline pkg/obs/driver.go:68]: pls append VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER and VolumeCapability_AccessMode_SINGLE_NODE_MULTI_WRITER

File: cmd/obs-csi-plugin/main.go (hunk 1 above)
Comment (deer-hang) [reply cmd/obs-csi-plugin/main.go:2]: done

File: pkg/obs/controllerserver.go (hunk 2 above)
Comment (deer-hang) [reply pkg/obs/controllerserver.go:2]: done

File: pkg/obs/driver.go (hunk 3 above)
Comment (deer-hang) [reply pkg/obs/driver.go:2]: done

File: pkg/obs/identityserver.go (hunk 4 above)
Comment (deer-hang) [reply pkg/obs/identityserver.go:2]: done

File: pkg/obs/nodeserver.go (hunk 5 above)
Comment (deer-hang) [reply pkg/obs/nodeserver.go:2]: done

File: pkg/obs/server.go (hunk 6 above)
Comment (deer-hang) [reply pkg/obs/server.go:2]: done

File: cmd/obs-csi-plugin/Dockerfile (hunk 7 above)
Comment (deer-hang) [reply cmd/obs-csi-plugin/Dockerfile:1]: done

File: cmd/obs-csi-plugin/build-obsfs.sh (hunk 8 above)
Comment (deer-hang) [reply cmd/obs-csi-plugin/build-obsfs.sh:1]: done

File: cmd/obs-csi-plugin/main.go (hunk 9 above)
Comment (deer-hang) [reply cmd/obs-csi-plugin/main.go:36]: done

File: cmd/obs-csi-plugin/main.go (hunk 10 above)
Comment (deer-hang) [reply cmd/obs-csi-plugin/main.go:83]: done

File: cmd/obs-csi-plugin/main.go (hunk 11 above)
Comment (deer-hang) [reply cmd/obs-csi-plugin/main.go:37]: done

File: pkg/obs/driver.go (hunk 13 above)
Comment (deer-hang) [reply pkg/obs/driver.go:57]: done

File: pkg/obs/driver.go (hunk 14 above)
Comment (deer-hang) [reply pkg/obs/driver.go:68]: done

**Pull request merged**